src/Application/Commands/ReturnErroneousExcessFees.php

```diff
@@ -0,0 +1,149 @@
+            $charge = $this->stripeClient->charges->retrieve($chargeId);
+
+            if (!$this->feeDiffers($donation, $charge) || $charge->status !== 'succeeded') {
+                continue;
+            }
+
+            $countOfDonationsWithDiscrepancy++;
+            $feeId = (string) $charge->application_fee;
+            $fee = $this->stripeClient->applicationFees->retrieve($feeId);
+
+            if ($fee->amount_refunded > 0) {
+                // Might be because the donation itself was fully refunded and the fee reversed. Carry on.
+                $output->writeln("Donation {$donation->getId()} has already had fee refunded");
+                continue;
+            }
+
+            $feeDifference = $fee->amount - $donation->getAmountToDeductFractional();
+
+            if ($donation->getCharityFee() !== $this->getCorrectFees($donation, $charge)->coreFee) {
+                $output->writeln("Donation {$donation->getId()} has a different fee than expected");
+                continue;
+            }
+
+            $countOfDonationsToChange++;
+            $sumOfFeeDifference = bcadd($sumOfFeeDifference, (string) ($feeDifference / 100), 2);
+
+            if ($mode === 'fix') {
+                $this->stripeClient->applicationFees->createRefund($feeId, [
+                    'amount' => $feeDifference,
+                    'metadata' => ['reason' => 'MAT-383 erroneous fee correction'],
+                ]);
+                $output->writeln("Refunded {$feeDifference} for {$donation->getUuid()}");
+            } else {
+                $output->writeln("Would refund {$feeDifference} for {$donation->getUuid()}");
+            }
+        }
+
+        $output->writeln(sprintf(
+            "Checked %d donations, found %d with differences, %d safe to correct, totalling %s",
+            $countOfDonationsChecked,
+            $countOfDonationsWithDiscrepancy,
+            $countOfDonationsToChange,
+            $sumOfFeeDifference,
+        ));
+
+        return 0;
+    }
+
+    /**
+     * Whether the actually charged `application_fee_amount` differs from the database Donation amount to deduct.
+     */
+    private function feeDiffers(Donation $donation, \Stripe\Charge $charge): bool
+    {
+         return $charge->application_fee_amount !== $donation->getAmountToDeductFractional();
+    }
+
+    private function getCorrectFees(Donation $donation, \Stripe\Charge $charge): Fees
+    {
+        /**
+         * @var array|Card|null $card
+         */
+        $card = $charge->payment_method_details?->toArray()['card'] ?? null;
+        if (is_array($card)) {
+            /** @var Card $card */
+            $card = (object)$card;
+        }
+
+        if (!$card) {
+            throw new \LogicException('Cannot continue with no card on charge');
+        }
+
+        $cardBrand = $card->brand;
+        $cardCountry = $card->country;
+
+        return Calculator::calculate(
```

Comment:
the result of this calculation will be the same as what we recorded on the donation in the database, won't it? I was going to ask if we should use that instead of recalculating, but I guess a better option is to use both, and throw if they're different, for an extra check?

Reply:
oh yes where you output `Donation {$donation->getId()} has a different fee than expected`. Makes sense. Could that throw or at least log an error instead of just outputting? We don't expect it to ever happen, do we?

Reply:
Sorry just saw this. I was thinking we would probably search all the output for it on a bigger run. Can make it an error on my next PR too.